HelpMyStreetFE/HelpMyStreetFE/Controllers/AccountController.cs

```diff
@@ -164,6 +172,44 @@ public async Task<IActionResult> LoadAwardsComponent(CancellationToken cancellat
             return ViewComponent("Awards", new { userID = user.ID, cancellationToken = cancellationToken });
         }
 
+        [Route("print-job-details")]
+        [HttpGet]
+        public async Task<IActionResult> PrintJobDetails(string j, CancellationToken cancellationToken)
+        {
+            var jobID = Base64Utils.Base64DecodeToInt(j);
+            User user = await _authService.GetCurrentUser(HttpContext, cancellationToken);
+            if (user == null)
+            {
+                return Redirect("/account/login?returnURL=/account/print-job-details?id=" + j);
+            }
+
+            return ViewComponent("JobDetail", new { JobID = jobID, User = user, Jobset = JobSet.UserAcceptedRequests, ToPrint = true});
+        }
+
+        [Route("email-job-details")]
+        [HttpGet]
+        public async Task<IActionResult> EmailJobDetails(string j, CancellationToken cancellationToken)
+        {
+            var jobID = Base64Utils.Base64DecodeToInt(j);
+            
+            User user = await _authService.GetCurrentUser(HttpContext, cancellationToken);
+            if (user == null)
```

Comment:
Again, this code will never be executed if there isn't a user in the session.  However, in this case we don't want a redirect to the login screen so follow the pattern from GetLoggedInStatus and decorate with:
`
        [AllowAnonymous]
        [AuthorizeAttributeNoRedirect]
`